### PR TITLE
[All] Remove ForceMask

### PR DESCRIPTION
**Pull request by fredroy**
https://github.com/sofa-framework/sofa/issues/1637

All in all, removes everything related to the ForceMask, which had been disabled (SOFA_ENABLE_MASK) for a long time.
It was also cauing bugs apparently to a lot of people.

[ci-depends-on https://github.com/sofa-framework/Flexible/pull/1]
[ci-depends-on https://github.com/sofa-framework/Compliant/pull/1]

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/SofaCore/src/sofa/core/Multi2Mapping.inl [hunk 1]
```diff
@@ -224,18 +212,15 @@ void Multi2Mapping<In1,In2,Out>::computeAccFromMapping(const MechanicalParams* m
...
-    for( unsigned i=0 ; i<this->fromModels1.size() ; ++i )
-        if( core::behavior::BaseMechanicalState* stateFrom = this->fromModels1[i]->toBaseMechanicalState() ) maskFrom1[i] = &stateFrom->forceMask;
-    maskFrom2.resize( this->fromModels2.size() );
-    for( unsigned i=0 ; i<this->fromModels2.size() ; ++i )
-        if( core::behavior::BaseMechanicalState* stateFrom = this->fromModels2[i]->toBaseMechanicalState() ) maskFrom2[i] = &stateFrom->forceMask;
-    maskTo.resize( this->toModels.size() );
-    for( unsigned i=0 ; i<this->toModels.size() ; ++i )
-        if (core::behavior::BaseMechanicalState* stateTo = this->toModels[i]->toBaseMechanicalState()) maskTo[i] = &stateTo->forceMask;
-        else this->setNonMechanical();
+    for (unsigned i = 0; i < this->toModels.size(); ++i)
```
Comment (alxbilger) [inline SofaKernel/modules/SofaCore/src/sofa/core/Multi2Mapping.inl]: range for loop

File: SofaKernel/modules/SofaCore/src/sofa/core/Multi2Mapping.inl [hunk 2]
```diff
@@ -224,18 +212,15 @@ void Multi2Mapping<In1,In2,Out>::computeAccFromMapping(const MechanicalParams* m
...
+    {
+        if (!this->toModels[i]->toBaseMechanicalState())
+        {
+            this->setNonMechanical();
```
Comment (alxbilger) [inline SofaKernel/modules/SofaCore/src/sofa/core/Multi2Mapping.inl:221]: break?
and what happens if one of the `toModels` is a `MechanicalState` but not the other?

Not related to this PR.

File: SofaKernel/modules/SofaCore/src/sofa/core/MultiMapping.inl [hunk 3]
```diff
@@ -122,13 +122,13 @@ type::vector<behavior::BaseMechanicalState*> MultiMapping<In,Out>::getMechTo()
...
 void MultiMapping<In,Out>::init()
 {
-    maskFrom.resize( this->fromModels.size() );
-    for( unsigned i=0 ; i<this->fromModels.size() ; ++i )
-        if( core::behavior::BaseMechanicalState* stateFrom = this->fromModels[i]->toBaseMechanicalState() ) maskFrom[i] = &stateFrom->forceMask;
-    maskTo.resize( this->toModels.size() );
-    for( unsigned i=0 ; i<this->toModels.size() ; ++i )
-        if( core::behavior::BaseMechanicalState* stateTo = this->toModels[i]->toBaseMechanicalState() ) maskTo[i] = &stateTo->forceMask;
-        else this->setNonMechanical();
+    for (unsigned i = 0; i < this->toModels.size(); ++i)
```
Comment (alxbilger) [inline SofaKernel/modules/SofaCore/src/sofa/core/MultiMapping.inl]: range for loop

File: SofaKernel/modules/SofaCore/src/sofa/core/behavior/MixedInteractionForceField.inl [hunk 4]
```diff
@@ -73,11 +73,10 @@ void MixedInteractionForceField<DataTypes1, DataTypes2>::addForce(const Mechanic
     if (mstate1 && mstate2)
     {
 
-            addForce( mparams, *fId[mstate1.get()].write()   , *fId[mstate2.get()].write()   ,
-                    *mparams->readX(mstate1), *mparams->readX(mstate2),
-                    *mparams->readV(mstate1), *mparams->readV(mstate2) );
+        addForce( mparams, *fId[mstate1.get()].write()   , *fId[mstate2.get()].write()   ,
```
Comment (alxbilger) [inline SofaKernel/modules/SofaCore/src/sofa/core/behavior/MixedInteractionForceField.inl]:
```suggestion
        addForce( mparams, *fId[mstate1.get()].write(), *fId[mstate2.get()].write(),
```

**Review by alxbilger — COMMENTED** [review]
Minor suggestions

File: SofaKernel/modules/SofaCore/src/sofa/core/Multi2Mapping.inl (hunk 2 above)
Comment (fredroy) [reply SofaKernel/modules/SofaCore/src/sofa/core/Multi2Mapping.inl:221]: I was thinking the same while doing this...

**Comment by guparan** [comment]
[ci-build][with-all-tests]

**Pull request merged**